azure-pipelines/acr-clean.yml

```diff
@@ -0,0 +1,64 @@
+pr: none
+trigger: none
+
+schedules:
+- cron: '0 0 1 * *'
+  always: true
+  branches:
+    include:
+    - main
+
+jobs:
+- job: cleanup
+  pool: sonicbld
+  timeoutInMinutes: 120
+  variables:
+    - group: Debian-Mirror-Common
+
+  steps:
+    - checkout: none
+    - script: |
+        curl -sL https://aka.ms/InstallAzureCLIDeb | sudo bash
+      displayName: install dependencies
+    - bash: |
+          set -ex
+          az login -u $(ApplicationId) --service-principal --tenant $(AzureTenant) -p "$ApplicationKey"
+          az account set --subscription 9355ef17-3aa2-493a-94ab-a43a9bf8cd70
+          az acr show -n sonicdev
+          sudo apt-get update
+          sudo apt install -y jq
+
+      env:
+        ApplicationKey: '$(ApplicationKey)'
+      displayName: az login
+    - bash: |
+        set -e
+
+        for registry in $(sonic_acrs); do
+            for repo in $(sonic_slaves); do
+                echo "$registry $repo"
+                rm -rf tmp
+                touch tmp
+                for branch in $(sonic_branches); do
+                    az acr repository show -n $registry --image $repo:$branch 2>/dev/null | jq .digest -r >> tmp || continue
+                    echo "    $repo:$branch $(tail -n 1 tmp)"
+                done
+
+                az acr repository show-tags -n $registry --repository $repo --orderby time_desc --output table > a
+                (( $(cat a | wc -l) < 100 )) && continue
+                tag=$(tail -n 10 a | head -n 1)
```

Comment:
Do  you want to skip first 100 lines?

```
tail -n +100
az acr repository show-tags -n $registry --repository $repo --orderby time_desc --output table | tail -n +100
```

Reply:
1. I want to keep latest 100 tags in each repo.
2. I want to delete at most 10 tags each run.